Add first draft of governance text as `governance.rst.txt`

slumnitz opened the pull request:
This is a first draft of possible governance guidelines for PyAL.
Conversations about governance structures started due to a possible NumFOCUS affiliation https://github.com/pysal/pysal/issues/1067

Main points:
* we will have a Benevolent Dictator for Life (currently @sjsrey) 
* together with a steering council (starting with current Sub-Package Maintainers)
* The Steering Council has a Chair who is responsible to keep all organisation documents up to date and potentially writes a little review about major updates yearly. 
* We have release managers for every Sub-Package who are the Maintainers and one release manager for the meta package
* There have to be at least two sub-package Maintainers

We could additionally decide on and include fixed reporting, review times if necessary (e.g. report on major updates all 6 months)

Please review, comment and add ideas!

slumnitz commented:
As a suggestion we could add all organisational Documents, e.g. CoC, Governance, Funding... under the heading `developer's guide`?

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+The Project¶
+------------
+
+PySAL, the Python spatial analysis library, is an open source cross-platform project (The Project). The goal of The Project is to develop open source software for geospatial data science in Python, and in particular the PySAL library and its Sub-Packages. The Software developed by The Project is released under the BSD (or similar) open source license, developed openly and hosted on public GitHub  ...
```
... PySAL library and its component modules (hereinafter referred to as "Sub-Packages").

sjsrey replied:
s/Sub-Packages/Packages/

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+The Project¶
+------------
+
+PySAL, the Python spatial analysis library, is an open source cross-platform project (The Project). The goal of The Project is to develop open source software for geospatial data science in Python, and in particular the PySAL library and its Sub-Packages. The Software developed by The Project is released under the BSD (or similar) open source license, developed openly and hosted on public GitHub  ...
```
The software developed by...

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+
+The Project is developed by a team of distributed developers, called Contributors. Contributors are individuals who have contributed code, documentation, designs or other work to The Project. Anyone can be a Contributor. Contributors can be affiliated with any legal entity or none. Contributors participate in the project by submitting, reviewing and discussing GitHub Pull Requests and Issues and  ...
+
+The Project Community consists of all Contributors and Users of The Project. Contributors work on behalf of and are responsible to the larger Project Community and we strive to keep the barrier between Contributors and Users as low as possible.
```
... all Contributors and users of the software developed by Contributors (hereinafter referred to as "Users") of The Project.

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+* Active Contribution
+* Institutional Neutrality
+
+Traditionally, Project leadership was provided by a BDFL (Sergio Rey) and a subset of Contributors and Sub-Package Maintainers, called Core Developers, whose active and consistent contributions have been recognized by their receiving “commit rights” to the Project GitHub repositories. In general all Project decisions are made through consensus among the Core Developers with input from the Communi ...
```
...by a BDFL (Benevolent Dictator for Life).

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+
+BDFL
+----
+The Project will have a BDFL (Benevolent Dictator for Life), who is currently Sergio Rey. As Dictator, the BDFL has the authority to make all final decisions for The Project. As Benevolent, the BDFL, in practice chooses to defer that authority to the consensus of the community discussion channels and the Steering Council (see below). It is expected, and in the past has been the case, that the BDF ...
```
~~ remove ~~ (Benevolent Dictator for Life)

jGaboardi replied:
... his/her...
- maybe we should either use more inclusive gender pronouns or stick with "they"

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+
+Sub-Package Maintainers
+-----------------------
+The Project is an umbrella for numerous cross-platform sub-packages. Sub-packages are maintained, managed and bound by the guidlines of the Sub-Package contract. 
```
guidelines

sjsrey replied:
s/Sub-//

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+
+The Council reserves the right to eject current Members, other than the BDFL, if they are deemed to be actively harmful to the project’s well-being, and attempts at communication and conflict resolution have failed.
+
+A list of current Steering Council Members is maintained at the page Current steering council and institutional partners.
```
...is maintained at Current Steering Council...

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+* Determining the content of a release in case there is no consensus on a particular change or feature.
+* Creating the release and announcing it on the relevant public channels.
+
+The responsibility of all sub-package resleases stays with the respective sub-package Maintainers.
```
Sub-Package releases

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+* Acknowledgement on the PySAL website and in talks.
+* Ability to acknowledge their own funding sources on the PySAL website and in talks.
+* Ability to influence the project through the participation of their Council Member.
+* Council Members invited to SciPy Developer Meetings.
```
Should this be PySAL Developer Meetings?

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+
+Document history
+----------------
+https://github.com/pysal/pysal.github.io/tree/master/_sources/governance.rst.txt
```
Should `governance.rst.txt` simply be `governance.rst` or `governance.md`?

jGaboardi replied:
@slumnitz Does this still need attention?

slumnitz replied:
Since the history should be pointing to this discussion/ commits and the changes we made to the existing document, it should be pointing here? We could also leave this out completely and only start a Document history once we decide the text needs changes in future?

jGaboardi commented:
> As a suggestion we could add all organisational Documents, e.g. CoC, Governance, Funding... under the heading `developer's guide`?

I'm not sure that `developer's guide` would the correct naming, but I do agree that we should one directory containing all the "admin"-type documents.

sjsrey commented on _sources/governance.rst.txt:
```diff
@@ -0,0 +1,169 @@
+The Project¶
+------------
+
+PySAL, the Python spatial analysis library, is an open source cross-platform project (The Project). The goal of The Project is to develop open source software for geospatial data science in Python, and in particular the PySAL library and its Sub-Packages. The Software developed by The Project is released under the BSD (or similar) open source license, developed openly and hosted on public GitHub  ...
```
s/Sub-Packages/Packages/

sjsrey commented:
> > As a suggestion we could add all organisational Documents, e.g. CoC, Governance, Funding... under the heading `developer's guide`?
> 
> I'm not sure that `developer's guide` would the correct naming, but I do agree that we should one directory containing all the "admin"-type documents.

Maybe a  heading like "Get Involved" or "Joining the Community" ?

slumnitz commented:
> Thanks for the work on this. I added some inline comments above.

> Here are some general thoughts:

> We should acknowledge our governance policies are adaptations of the Jupyter (and/or others we researched in preparation)

This happened in the Acknowledgements sections right at the bottom

> We may also want to use "[Pp]ackage" rather than "[Ss]ub-[Pp]ackage" throughout.

> If we refer to PySAL as the project, rather than a package, this leaves package available rather than having to use subpackage.

I was hoping the text addresses both, the governance of `pysal` as a package and all sub-packages under the umbrella of the PySAL project, hence the differentiation. I also think we should keep it consistent with other names we introduced earlier like the sub-package contract, to avoid confusion. But I can reformulate it a little so it might get clearer?

slumnitz commented:
@weikang9009 sure no problem! I think it still is to be determined where to put it though. 

@jGaboardi and @weikang9009 would it be possible to lead to separate pages for `CoC`, `Governance`, `Building Documentation`, `Submodule Contract` and `Submodule Template` under the `Getting Started` -> `For Developers` heading? So to say not linking to `CoC` or `Governance` in the upper index but under the `For Developers` Section?

weikang9009 commented:
> @jGaboardi and @weikang9009 would it be possible to lead to separate pages for `CoC`, `Governance`, `Building Documentation`, `Submodule Contract` and `Submodule Template` under the `Getting Started` -> `For Developers` heading? So to say not linking to `CoC` or `Governance` in the upper index but under the `For Developers` Section?

@slumnitz I like your idea of reorganizing the [`For Developers` section](http://pysal.org/getting_started.html#for-developers) as we are adding more contents which make it difficult for users to navigate if all the contents are on the same webpage.

slumnitz commented:
@sjsrey @jGaboardi have a look at the changes I made especially in 'The Project' section. If this looks good we can integrate it into the website in another PR. I noticed things we still have to do or decide on whether this is necessary or not:

- [ ] vote on Steering Committee Members
- [ ] vote on the Council Chair
- [ ] add a List of Members and Chair to the website (or start at least an empty list we can fill later on)

jGaboardi commented:
@slumnitz 

 - Is there any standard number of Steering Council Members for other projects? I am just wondering if there is some minimum number of people who need to be on the council.

jGaboardi commented on _sources/governance.rst.txt:
```diff
@@ -101,17 +101,17 @@ The Chair will be appointed by the Steering Council. The Chair can stay on as lo
 
 Release Manager
```
@slumnitz @sjsrey Do we also need to decide on a Release Manager or does this automatically fall to the BDFL?

slumnitz replied:
I would suggest to keep it separate appointments, but not exclude that someone can have multiple appointments?

sjsrey replied:
For the meta package, the one of the maintainers could be the release manager?

slumnitz commented:
> @slumnitz
> 
> * Is there any standard number of Steering Council Members for other projects? I am just wondering if there is some minimum number of people who need to be on the council.

There is no standard number, every project is different. For our project I would suggest quite a few, since we have many connected packages.

sjsrey commented:
> @sjsrey @jGaboardi have a look at the changes I made especially in 'The Project' section. If this looks good we can integrate it into the website in another PR. I noticed things we still have to do or decide on whether this is necessary or not:
> 
>     * [ ]  vote on Steering Committee Members

This would be for new members correct? I seems as if we are going with the initial composition being based on existing maintainers.

I wonder if it makes sense to have a Steering Council/Committee (we use both terms in the current document, we should fix this).  Maybe a steering council of say 5 members, nominated by the maintainers? Voted on by maintainers and core devs at a dev meeting?

> 
>     * [ ]  vote on the Council Chair
> 

Voted by the members of the council, once it is formed.

Once the
>     * [ ]  add a List of Members and Chair to the website (or start at least an empty list we can fill later on)

Yes. This should be listed.

slumnitz commented:
@sjsrey sounds great! if you create an empty repo I am happy to move and update everything.

sjsrey commented:
> @sjsrey sounds great! if you create an empty repo I am happy to move and update everything.

https://github.com/pysal/governance

jGaboardi commented:
@sjsrey Would you also like me to migrate the Code of Conduct stuff into the [`code_of_conduct`](https://github.com/pysal/code_of_conduct) repo, merge into the [`README.md`](https://github.com/pysal/code_of_conduct/blob/master/README.md), and close out the other PRs in pysal/pysal.site (pysal/pysal.site#27) and pysal.github.io (pysal/pysal.github.io#76, pysal/pysal.github.io#77)?

jGaboardi commented:
Closing as this is address by the addition of [`pysal/governance`](https://github.com/pysal/governance).
